airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceMetadataQuerier.kt

```diff
@@ -0,0 +1,174 @@
+/* Copyright (c) 2024 Airbyte, Inc., all rights reserved. */
+package io.airbyte.integrations.source.mysql
+
+import io.airbyte.cdk.check.JdbcCheckQueries
+import io.airbyte.cdk.command.SourceConfiguration
+import io.airbyte.cdk.discover.Field
+import io.airbyte.cdk.discover.JdbcMetadataQuerier
+import io.airbyte.cdk.discover.MetadataQuerier
+import io.airbyte.cdk.discover.SystemType
+import io.airbyte.cdk.discover.TableName
+import io.airbyte.cdk.discover.UserDefinedType
+import io.airbyte.cdk.jdbc.DefaultJdbcConstants
+import io.airbyte.cdk.jdbc.JdbcConnectionFactory
+import io.airbyte.cdk.read.SelectQueryGenerator
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Primary
+import jakarta.inject.Singleton
+import java.sql.ResultSet
+import java.sql.Statement
+
+private val log = KotlinLogging.logger {}
+
+/** Delegates to [JdbcMetadataQuerier] except for [fields]. */
+class MysqlSourceMetadataQuerier(
+    val base: JdbcMetadataQuerier,
+) : MetadataQuerier by base {
+
+    override fun fields(
+        streamName: String,
+        streamNamespace: String?,
+    ): List<Field> {
+        val table: TableName = findTableName(streamName, streamNamespace) ?: return listOf()
+        if (table !in base.memoizedColumnMetadata) return listOf()
+        return base.memoizedColumnMetadata[table]!!
+            .map { c: JdbcMetadataQuerier.ColumnMetadata ->
+                val udt: UserDefinedType? =
+                    when (c.type) {
+                        is SystemType -> allUDTsByFQName[c.type.typeName]
+                        else -> null
+                    }
+                c.copy(type = udt ?: c.type)
+            }
+            .map { Field(it.label, base.fieldTypeMapper.toFieldType(it)) }
+    }
+
+    val allUDTsByFQName: Map<String, UserDefinedType> by lazy {
+        val otherUDTs: List<UserDefinedType> = base.memoizedUserDefinedTypes
+        val allUDTs: List<UserDefinedType> = otherUDTs
+        allUDTs.reversed().associateBy { "${it.schema}.${it.typeName}" }
+    }
```

Comment:
You might not need all this UDT logic. This was in oracle only to support array types. From what I heard, mysql doesn't have them.